stuff/cron/testing/test_plagiarism_detector.py

```diff
@@ -0,0 +1,267 @@
+    verdict: str = "AC"
+    ttype: str = "test"
+    no_of_users: int = 3  # we can increase the number of user of further testing
+    no_of_problems: int = 3  # same as above case
+    GUID: List[str] = []
+    # contest = Contest()
+
+    # create Problemset for contest
+    with dbconn.cursor() as cur:
+        cur.execute(
+            '''
+                    INSERT INTO `Problemsets`
+                        (`acl_id`, `scoreboard_url`, 
+                        `scoreboard_url_admin`)
+                    VALUES
+                        (%s, %s, %s);
+                    ''', (
+                acl_id,
+                scoreboard_url,
+                scoreboard_url_admin,
+            ))
+        problemset_id: int = cur.lastrowid
+    dbconn.conn.commit()
+
+    # add 3 problems to problemset
+    for problem in range(1, no_of_problems + 1):
+        version = ''.join(random.choices(string.ascii_letters, k=40))
+        with dbconn.cursor() as cur:
+            cur.execute(
+                '''
+                INSERT INTO `Problemset_Problems`
+                (`problemset_id`, `problem_id`, `version`)
+                VALUES
+                (%s, %s, %s)
+                ''', (
+                    problemset_id,
+                    problem,
+                    version,
+                ))
+    dbconn.conn.commit()
+
+    # create a contest
+    with dbconn.cursor() as cur:
+        cur.execute(
+            '''
+                    INSERT INTO `Contests`
+                    (`title`, `alias`,
+                        `description`, 
+                    `start_time`, `finish_time`,
+                        `check_plagiarism`, `problemset_id`,
+                        `acl_id`
+                    )
+                    VALUES
+                    (%s, %s, %s, %s, %s, %s, %s, %s)
+                    ''', (
+                alias,
+                alias,
+                description,
+                start_time,
+                finish_time,
+                check_plagiarism,
+                problemset_id,
+                acl_id,
+            ))
+        Contest.contest_id = cur.lastrowid
+    dbconn.conn.commit()
+
+    # add problemset to contest
+    with dbconn.cursor() as cur:
+        cur.execute(
+            '''
+                UPDATE `Problemsets`
+                SET `contest_id` = (
+                    SELECT `contest_id` FROM
+                    `Contests` 
+                    WHERE `alias` = %s
+                )
+                WHERE problemset_id = %s;
+                    ''', (
+                alias,
+                problemset_id,
+            ))
+    dbconn.conn.commit()
+
+    # setting foreign key check = 0 because of dependencies
+    with dbconn.cursor() as cur:
+        cur.execute('''
+                SET foreign_key_checks = 0;
+                TRUNCATE TABLE Plagiarisms;
+                TRUNCATE TABLE Submission_Log;
+                SET foreign_key_checks = 1;
+                ''')
```

Comment:
```suggestion
        cur.execute('SET foreign_key_checks = 0;')
        cur.execute('TRUNCATE TABLE Plagiarisms;')
        cur.execute('TRUNCATE TABLE Submission_Log;')
        cur.execute('SET foreign_key_checks = 1;')
```

ooh sorry, this is what i meant: share the `with`, but unshare the `cur.execute`.

Reply:
Done.